lib/shanty/plugin.rb

```diff
@@ -31,17 +32,21 @@ def self.tags(*args)
     end
 
     def self.option(option, default = nil)
-      config[name][option] = default if config[name][option].nil?
+      plugin_options[option] = default
     end
 
     def self.options
-      config[name]
+      plugin_options.merge(config[name].to_h).to_ostruct
```

Comment:
Better off inlining `self.plugin_options` here like the other methods of this ilk:

```ruby
def self.options
  (@options ||= {}).merge(config[name].to_h).to_ostruct
end
```